### PR TITLE
11 chat age add prompt for translate profanity

### PR DESCRIPTION
**Pull request by MrBennySolomon**

### CODE REVIEW

File: server/api/translation/openAI.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const init = () => {
+  const config = new Configuration({
+    apiKey: process.env.OPEN_AI_API_KEY,
+  });
+  return new OpenAIApi(config);
+};
+
+const openAiInit = async (openai, prompt) => {
+  return await openai.createCompletion({
+    model: "text-davinci-003",
```
Comment (yishain11) [inline server/api/translation/openAI.js:20]: use chatGtp model

File: server/api/translation/openAI.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    prompt: prompt,
+    max_tokens: 2048,
+    temperature: 1,
+  });
+};
+
+const runPrompt = async (question, from, to) => {
+  const openai = init();
+  const prompt = `is the text profanity: <text>${question}</text> answer only "true" or "false"`;
```
Comment (yishain11) [inline server/api/translation/openAI.js:29]: "is ther profanity in this text"

File: server/api/translation/openAI.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  const isProfanity = await openAiInit(
+    openai,
+    `is the text profanity? answer only "true" or "false" : ${prompt}`
```
Comment (yishain11) [inline server/api/translation/openAI.js:32]: you are using "is the text profanity? answer only "true" or "false" " twice, once in text and once in promt var

File: server/api/translation/openAI.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  );
+  if (isProfanity.data.choices[0].text.toLowerCase().includes("true")) {
+    return "Profanity";
```
Comment (yishain11) [inline server/api/translation/openAI.js:35]: return the msg described in the issue

File: server/api/user/user.controllers.js [hunk 5]
```diff
@@ -1 +1,15 @@
-//test
+import asyncHandler from "../../middleware/asyncHandler.js";
+import User from "./user.js";
+
+export const registerUser = asyncHandler(async (req, res, next) => {
+  console.log("inside user controller");
```
Comment (yishain11) [inline server/api/user/user.controllers.js:5]: remove clog

File: server/api/translation/openAI.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const openAiInit = async (openai, prompt) => {
+  return await openai.createChatCompletion({
+    model: "gpt-3.5-turbo",
+    messages: [{ role: "user", content: prompt }],
+  });
+};
+
+const runPrompt = async (prompt, from, to) => {
```
[the remark this answers is not in the document]
Comment (MrBennySolomon) [reply server/api/translation/openAI.js:26]: 1.the user write message
2.we send a post request to .../arabrew/translation/:user1/:user2/:original_lang/:target_lang
3. the message will be in the req.body
4. i handle it in the translation.controllers.js
5. and there i import runPrompt from "./openAI.js";
6. so of course i'm sending the message as parameter to the runPrompt function

File: server/api/translation/translation.controllers.js [hunk 7]
```diff
@@ -1 +1,39 @@
-//test
+import asyncHandler from "../../middleware/asyncHandler.js";
+import runPrompt from "./openAI.js";
+
+export const translateMsg = asyncHandler(async (req, res, next) => {
+  let translated = await runPrompt(
+    req.body.data,
```
[the remark this answers is not in the document]
Comment (MrBennySolomon) [reply server/api/translation/translation.controllers.js:7]: i'm getting the message from the user, this is what the app should be doing
and then i check if it's profanity, if yes i return profanity message,
if it is not profanity message, i translate the message, and save to the DB

File: server/server.js [hunk 8]
```diff
@@ -38,7 +42,9 @@ wss.on("connection", (ws) => {
   };
 });
 
-const PORT = process.env.PORT || 5005;
+app.use("/arabrew", routes);
```
[the remark this answers is not in the document]
Comment (MrBennySolomon) [reply server/server.js:45]: /arabrew is the prefix that taufiq, the team leader build,
i can change it ti /api if you want

**Pull request merged**